app.ts

```diff
@@ -0,0 +1,216 @@
+window.onload = () => {
+
+    let numColumns: number;
+    let columns: any;
+    let recordCount: number;
+    let startIndex = 0; // Default value = 0;
+    let tableRecordCount = 20; // Default value = 20;
+    let timer: number;
+
+    // Canvas:
+    let canvas = document.createElement("div");
+    canvas.id = "canvas";
+    $("body").append(canvas);
+
+    // Buttons and fields:
+    let searchbtn = document.createElement("button");
+    searchbtn.innerText = "Search";
+
+    let nextbtn = document.createElement("button");
+    nextbtn.innerText = "Next Page";
+
+    let prevbtn = document.createElement("button");
+    prevbtn.innerText = "Previous Page";
+
+    let fromField = document.createElement("input");
+    fromField.placeholder = "from";
+
+    // Window resizing (using a debouncing method):
+    window.onresize = () => {
+        let time = 100;
+        clearInterval(timer);
+        timer = setInterval(function () {
+            clearInterval(timer);
+            adjustTableRecordCount();
+            let end = tableRecordCount - 1;
+            if (recordCount < startIndex + tableRecordCount) {
+                startIndex = recordCount - tableRecordCount;
+            }
+            getRecords(startIndex, startIndex + end);
+        }, time);
+    }
+    
+    // on clicks:
+    searchbtn.onclick = function() {
+        let from = parseInt(fromField.value, 10);
+        // validate field:
+        if (validate(from)) {
+            let end = tableRecordCount - 1;
+            startIndex = from;
+            if (recordCount < startIndex + tableRecordCount) {
+                startIndex = recordCount - tableRecordCount;
+            }
+            getRecords(startIndex, startIndex + end);
+        }
+    };
+
+    nextbtn.onclick = function() {
+        let end = tableRecordCount - 1;
```

Comment:
Considering that you only use this as an offset right at the end (when you call _getRecords_), is it entirely necessary to declare this as a variable of its own?